writers/streamingbatchwriter/unimplemented.go

```diff
@@ -27,3 +27,9 @@ func (UnimplementedDeleteStale) DeleteStale(_ context.Context, ch <-chan *messag
 	}
 	return fmt.Errorf("DeleteStale: %w", plugin.ErrNotImplemented)
 }
+
+type UnimplementedDeleteRecordsBatch struct{}
```

Comment:
```suggestion
type UnimplementedDeleteRecords struct{}
```
would be a better name.